cmd/dockerd/daemon.go

```diff
@@ -227,13 +228,18 @@ func (cli *DaemonCli) start(opts daemonOptions) (err error) {
 
 	name, _ := os.Hostname()
 
+	// Use a buffered channel to pass changes from store watch API to daemon
+	// A buffer allows store watch API and daemon processing to not wait for each other
+	watchStream := make(chan *swarmapi.WatchMessage, 32)
```

Comment:
No need to change but wondering why you chose to pass in a channel and handle this in dockerd instead of defining a `LogEvent` interface in cluster, that would be called to process events.